design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,128 @@
+
+#### Requirements
+
+The CLI design should answer the following requirements:
+
+*Core functionality*
+
+1. For each platform, we should support *init*, *login*, *authenticate*, *load policy*, and *fetch secrets* functionalities. These functionalities are must-haves but we should not be limited to providing just these.
+
+*Deployment*
+
+1. Provide a manual and automated way to deploy the CLI
+2. Require a maximum of 2 steps
+3. ***Recommended:*** cross-platform deployment (Linux, Windows) should all have the same flow
+
+## Research
+
+[//]: # "Elaborate on the issue you are writing a solution for"
+
+Currently, we provide 3 CLIs that can be used to interface with Conjur, Ruby, Golang, Python. We will explore and differentiate each in terms of their functionality, quality, UX, testing, security and what it is required to deploy them.
+
+|            | Functionality                                                |
+| ---------- | ------------------------------------------------------------ |
+| Ruby CLI   | Most expressive coverage                                     |
+| Golang CLI | Implementation is limited with poor architectural design. We could salvage what is currently in-place but it was recommended by the contributor to rethink the architecture and even perform a complete rewrite. |
+| Python CLI | More or less at level of Golang CLI in terms of functionality coverage but with more intuitive commands. For example, in Golang `conjur-cli variable value <variable_id>` and in Python `conjur-cli variable get <variable_id>` . |
+
+The following are mappings of functionalities for both Golang and Python CLIs.
+
+![golang-cli-mapping](../images/golang-cli-mapping.png)
+
+![python-cli-mapping](../images/python-cli-mapping.png)
+
+
+
+|            | Quality                                                 |
+| ---------- | ------------------------------------------------------- |
+| Ruby CLI   | Widely used<br />***Status***: Stable                   |
+| Go CLI     | Not "community" level certified <br />***Status***: POC |
+| Python CLI | ***Status***: Community/alpha level                     |
+
+|            | Deployment                                                   |
+| ---------- | ------------------------------------------------------------ |
+| Ruby CLI   | Options:<br />1. Deploy to a Docker container (not applicable for our use-case)<br />2. Deploy using a very specific version of Ruby<br />Which means that the CLI is not plug-and-play and has prerequisites of either Docker or Ruby. |
+| Go CLI     | The CLI is compiled down to a binary. A different binary is need to be compiled for each platform but no other requirements or installers are needed to launch the Go CLI. |
+| Python CLI | Compiled down to a single binary without the need for Python or other prerequisites to be installed on machine. |
+
+|            | Testing                                                      |
```

Comment:
Need to understand how we are going to test it, and on which platforms?
Do we have basic automation on each api today?